server/src/main/java/org/apache/iotdb/db/common/RetryCounter.java

```diff
@@ -0,0 +1,129 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.db.common;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+
+public class RetryCounter {
+
+  private static final Logger logger = LoggerFactory.getLogger(RetryCounter.class);
+
+  private RetryConfig retryConfig;
+  private int attempts;
+
+  public RetryCounter(RetryConfig retryConfig) {
+    this.attempts = 1;
+    this.retryConfig = retryConfig;
+  }
+
+  public void sleepToNextRetry() throws InterruptedException {
+    int attempts = getAttempts();
+    long sleepTime = getBackoffTime();
+    logger.trace("Sleeping {} ms before retry {}", sleepTime, attempts);
+    retryConfig.getTimeUnit().sleep(sleepTime);
+    useRetry();
+  }
+
+  public boolean shouldRetry() {
+    return attempts < retryConfig.getMaxAttempts();
+  }
+
+  public RetryConfig getRetryConfig() {
+    return retryConfig;
+  }
+
+  public int getAttempts() {
+    return attempts;
+  }
+
+  public long getBackoffTime() {
+    return this.retryConfig.backoffPolicy.getBackoffTime(this.retryConfig, getAttempts());
+  }
+
+  private void useRetry() {
+    attempts++;
+  }
+
+  public static class RetryConfig {
+    private int maxAttempts;
+    private long sleepInterval;
+    private TimeUnit timeUnit;
+    private BackoffPolicy backoffPolicy;
+    private float jitter;
+
+    private static final BackoffPolicy DEFAULT_BACKOFF_POLICY = new ExponentialBackoffPolicy();
+
+    public RetryConfig() {
+      maxAttempts = 1;
+      sleepInterval = 100;
+      timeUnit = TimeUnit.MILLISECONDS;
+      backoffPolicy = DEFAULT_BACKOFF_POLICY;
+      jitter = 0.0f;
+    }
+
+    public RetryConfig setMaxAttempts(int maxAttempts) {
+      this.maxAttempts = maxAttempts;
+      return this;
+    }
+
+    public RetryConfig setSleepInterval(long sleepInterval) {
+      this.sleepInterval = sleepInterval;
+      return this;
+    }
+
+    public int getMaxAttempts() {
+      return maxAttempts;
+    }
+
+    public long getSleepInterval() {
+      return sleepInterval;
+    }
+
+    public TimeUnit getTimeUnit() {
+      return timeUnit;
+    }
+
+    public float getJitter() {
+      return jitter;
+    }
+  }
+
+  private static long addJitter(long interval, float jitter) {
+    long jitterInterval = (long) (interval * ThreadLocalRandom.current().nextFloat() * jitter);
+    return interval + jitterInterval;
+  }
+
+  public static class BackoffPolicy {
+    public long getBackoffTime(RetryConfig config, int attempts) {
+      return addJitter(config.getSleepInterval(), config.getJitter());
+    }
+  }
+
+  public static class ExponentialBackoffPolicy extends BackoffPolicy {
+    @Override
+    public long getBackoffTime(RetryConfig config, int attempts) {
+      long backoffTime = (long) (config.getSleepInterval() * Math.pow(2, attempts));
```

Comment:
`long backoffTime = (long) (config.getSleepInterval() * Math.pow(2, attempts));`

The above code line seems not used?

Reply:
thanks @neuyilan , this code is called in sleepUntilNextRetry - > getBackoffTime  method...